### ISSUE TITLE
Docs: bench::some_id unique or not?

### ISSUE THREAD

**Issue by peter-kehl** [iai-callgrind/iai-callgrind#169]
Hi & thank you for `iai-callgrind`.

Please clarify whether the ID part in `#[bench::some_id(value)]` is unique, and if so, in what scope/limit and where can it be repeated again and compared in the same group.
- https://docs.rs/iai-callgrind/latest/iai_callgrind/attr.library_benchmark.html > "The bench attribute consist of the attribute name itself, an unique id after ::"
- https://docs.rs/iai-callgrind/latest/iai_callgrind/macro.library_benchmark_group.html > "... are compared with each other as long as the ids (the part after the :: in #[bench::id(...)]) match"
- https://github.com/iai-callgrind/iai-callgrind/blob/main/README.md#comparing-benchmark-functions > 
  ```
  #[library_benchmark]
  #[bench::case_3(vec![1, 2, 3])]
  ...
  #[library_benchmark]
  #[bench::case_3(vec![3, 2, 1])]
  ```

(Update:) Or, does it mean that `bench::` ID(s) of/for the same `#[library_benchmark]` must be unique, but then another `#[library_benchmark]` may have some/all ID(s) same and they are independent. So, IDs are separate/isolated between `#[library_benchmark]`-s?

**Comment by gamma0987** [comment]
Hi. 

> (Update:) Or, does it mean that bench:: ID(s) of/for the same #[library_benchmark] must be unique, but then another #[library_benchmark] may have some/all ID(s) same and they are independent. So, IDs are separate/isolated between #[library_benchmark]-s?

Yes. The id has to be unique within the same `#[library_benchmark]` (at benchmark function level), so your example

```rust
#[library_benchmark]
#[bench::case_3(vec![1, 2, 3])]
...
#[library_benchmark]
#[bench::case_3(vec![3, 2, 1])]
```

works even when both benchmark functions are added to the same group. But, you can't do this (Doing it results in a compiler error message):

```rust
#[library_benchmark]
#[bench::case_3(vec![1, 2, 3])]
#[bench::case_3(vec![3, 2, 1])]
```

For example, we exploit that to organise the comparison between the benchmark functions and `bench`es with the same ids at group level (`compare_by_id = true`).

**Comment by gamma0987** [comment]
Closing this as solved with #173
